### ISSUE TITLE
[GKE v2] Imported Cluster Imported Flag is changed after sync

### ISSUE THREAD

**Issue by westlywright** [rancher/rancher#32072]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Create a GKE cluster in GKE Console that will be imported.
In Rancher UI import the GKE cluster.
Allow the sync to occur between rancher and gke.

**Result:**
Once the sync is complete, view the cluster data in the api and see that in upstreamsepec imported is no longer true. This causes issues with syncing because the upstreamspec is the source of truth for the UI to sync the changes down to the empty GKE config when editing. 

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
- Installation option (single install/HA):

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
(paste the output here)
```

- Docker version (use `docker version`):

```
(paste the output here)
```

**Comment by westlywright** [comment]
Duplicate of https://github.com/rancher/rancher/issues/32072
